### PR TITLE
[WIP][Resource] Make api response codes configurable 

### LINKED ISSUE #5222: [RFC] A way to change response code and/or body of ResourceController::updateAction
Right now the `updateAction` is forcing a `204` and provides you with an empty `View` object (no data in it). Before you had the resource with it and you could for example override the `ViewHandler` and manipulate the response to be `200` with the serialized resource with it. 

For my API that I build around all Sylius resources I am in need to have the whole resource serialized in the payload when you update it, instead of a `HTTP 204 No Content`. 

Before this change happend, I achieved it by doing the following in a decorated view handler:

``` php
/**
 * {@inheritdoc}
 */
public function handle(RequestConfiguration $requestConfiguration, View $view)
{
    if (Response::HTTP_NO_CONTENT === $view->getStatusCode() && null !== $view->getData()) {
        $view->setStatusCode(Response::HTTP_OK);
    }

    return $this->viewHandler->handle($requestConfiguration, $view);
}
```

It is not the cleanest solution, but it worked for me. Right now I am stuck since the whole data property of the view is `null`.

This problem was introduced by PR https://github.com/Sylius/Sylius/pull/5161.

### PR DESCRIPTION
**Pull request by koemeet**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | maybe |
| Related tickets | fixes #5222 |
| License | MIT |

> I made one big change and that is using `array_replace_recursive` instead of `array_merge` for the default parameters and the passed in api parameters. This was needed so the default configuration `response_code` would stay intact when overridden inside a single route.

You can basically change the response code of every single action inside the resource controller.
### Usage

You can change the default response codes globally by doing:

``` yml
sylius_resource:
    settings:
        response_code:
            create: 201
            update: 200
```

You can then override specific response codes again on a per-route basis:

``` yml
sylius_api_some_resource:
    #...
    defaults:
        _sylius:
            response_code:
                create: 204 
```

You can override response codes for the following actions:
- create
- update
- delete 
- applyStateMachineTransition
- enable
- disable
- moveUp
- moveDown

Please let me know what you guys think about such a solution! Would love to hear any ideas for improvements on this 😄

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php [hunk 1]
```diff
@@ -92,6 +92,23 @@ public function isHtmlRequest()
...
+     * @return integer
+     */
+    public function getResponseCode()
+    {
+        $responseCode = $this->parameters->get('response_code');
+
+        if (is_array($responseCode)) {
+            $controller = $this->request->attributes->get('_controller');
+            if (preg_match('/^.+\:(.+)Action$/i', $controller, $matches)) {
+                return $responseCode[$matches['1']];
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php]: Why did you use a number in `'`? Also, why it is always 1?

Comment (steffenbrem) [inline src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php]: It is faster to quote array indexes.

Comment (steffenbrem) [inline src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php]: I will improve this part and also want to cover different cases in phpspec. Not even sure if I want to even use a reg ex.

**Comment by steffenbrem** [comment]
@lchrusciel Yeah indeed, it's still in a wip and wanted to show it early to showcase the configuration idea etc. code-wise this is not finished yet.

File: src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php (hunk 1 above)
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php]: It was my second question, is it needed here. Regex are useful, but usually hard to maintain.

**Pull request closed without merging**